### PR TITLE
[JENKINS-57023] Retain only deprecated global library functionality

### PR DESCRIPTION
**Pull request by jglick**
Complements https://github.com/jglick/pipeline-groovy-lib-plugin/pull/5 ⇒ https://github.com/jenkinsci/pipeline-groovy-lib-plugin/commit/4b687c23611b07efa26d0a11525912aaa56af606

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -88,56 +87,18 @@
 
         <!-- required plugins -->
         <dependency>
-            <groupId>org.jenkins-ci.plugins.workflow</groupId>
-            <artifactId>workflow-api</artifactId>
+            <groupId>io.jenkins.plugins</groupId>
+            <artifactId>pipeline-groovy-lib</artifactId>
+            <version>999999-SNAPSHOT</version> <!-- TODO -->
```
Comment (jglick) [inline pom.xml:92]: Not possible to get an incremental deployment from the upstream PR until it has been accepted by Hosting.

File: pom.xml [hunk 2]
```diff
@@ -88,56 +87,18 @@
...
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>git-server</artifactId>
         </dependency>
-        <dependency>
-            <groupId>org.jenkins-ci.plugins</groupId>
-            <artifactId>scm-api</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>org.jenkins-ci.plugins</groupId>
-            <artifactId>cloudbees-folder</artifactId>
```
Comment (jglick) [inline pom.xml:112]: Some more `mvn dependency:analyze`.

File: README.md [hunk 3]
```diff
@@ -1,12 +1,5 @@
...
-This plugin adds that functionality by allowing you to create “shared library script” SCM repositories.
-It can be used in two modes:
-
--   A legacy mode in which there is a single Git repository hosted by Jenkins itself, to which you may push changes
--   A more general mode in which you may define libraries hosted by any SCM in a location of your choice.
-
-Comprehensive user documentation can be found [in the Pipeline chapter of the User Handbook](https://jenkins.io/doc/book/pipeline/shared-libraries/).
+This plugin is deprecated.
+It implements a legacy mode of defining Groovy libraries in which there is a single Git repository hosted by Jenkins itself, to which you may push changes.
+Instead use **Pipeline: Groovy Libraries**.
```
Comment (dwnusbaum) [inline README.md:5]: Maybe turn this into a link? Also the display name is "Pipeline: Shared Groovy Libraries":

```suggestion
Instead use [Pipeline: Shared Groovy Libraries](https://plugins.jenkins.io/pipeline-groovy-lib/).
```

**Review by dwnusbaum — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (jglick) [reply README.md:5]: I meant to get rid of the _shared_ part (which is redundant for a _library_: if it were not shared, why would you need a library?). I will do the link.

Comment (dwnusbaum) [reply README.md:5]: Yes I agree that "Shared" is redundant, but from a consistency standpoint, existing documentation uses the term "Shared Libraries", e.g. https://www.jenkins.io/doc/book/pipeline/shared-libraries/, and https://github.com/jglick/pipeline-groovy-lib-plugin/pull/5 does not change the display name.

**Review by car-roll — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (jglick) [reply pom.xml:92]: (currently waiting on https://github.com/jenkinsci/pipeline-groovy-lib-plugin/pull/2)

**Pull request merged**